### PR TITLE
Add CODE-OF-CONDUCT.md

### LINKED ISSUE #18820: repo is missing a Code of Conduct statement
This repo should include a Code of Conduct statement.

### PR DESCRIPTION
**Pull request by mspncp**
Fixes #18820

### CODE REVIEW

**Review by paulidale — DISMISSED** [review]

**Review by levitte — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Code of Conduct
+===============
+
+The OpenSSL [Code of Conduct] is pulished on the project's website.
```
Comment (t8m) [inline CODE_OF_CONDUCT.md:4]: Typo: published

**Review by t8m — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (mspncp) [reply CODE_OF_CONDUCT.md]: Ups, good catch! Typos in the middle of a word are the most dangerous all. The human eye tends to see what it wants to see.

Comment (mspncp) [reply CODE_OF_CONDUCT.md]: I [force-pushed](https://github.com/openssl/openssl/compare/318eee3d3fdbbea853ef7adc36674510fadc0bc7..4b9d2923b31090db732f4b643a62f4a860367579) the correction.

**Review by levitte — APPROVED** [review]

**Comment by tom-cosgrove-arm** [comment]
(minor comment) None of the files in the top-level of the repo use `_` in the filename, and several of the `.md` files use `-` to separate words in their filenames. However, since the `_` is used instead of spaces, this could well be reasonable (in the other `.md` filenames, the `-`s are representing punctuation. As I said, just a comment to make sure this is intended.

**Review by t8m — APPROVED** [review]

**Comment by mspncp** [comment]
> (minor comment) None of the files in the top-level of the repo use `_` in the filename, and several of the `.md` files use `-` to separate words in their filenames. However, since the `_` is used instead of spaces, this could well be reasonable (in the other `.md` filenames, the `-`s are representing punctuation). As I said, this just a comment to make sure this is intended.

Good point. I chose the underscore because it was the filename recommended by GitHub's help. Personally, I would prefer underscores for all 'capital' markdown files in the root directory, but that would have to be decided and fixed independently. So for the sake of consistency, I'll rename the file from `CODE_OF_CONDUCT.md` to `CODE-OF-CONDUCT.md` for the moment.

I checked in the Web Interface that GitHub recognizes that filename as well, as the screenshot shows:


grafik

**Review by t8m — APPROVED** [review]
The name was OK with underscores too but this one is fine as well.

**Comment by mspncp** [comment]
> The name was OK with underscores too but this one is fine as well.

@t8m agreed. It' just the obvious inconsistency with the names of the other top-level markdown files which is disturbing.

@openssl/otc: what is your opinion about replacing dashes with underscores (in a followup pull request), as suggested by @tom-cosgrove-arm ?

**Dashes**
```
ACKNOWLEDGEMENTS.md
AUTHORS.md
CHANGES.md
CODE-OF-CONDUCT.md
CONTRIBUTING.md
FAQ.md
HACKING.md
INSTALL.md
NEWS.md
NOTES-ANDROID.md
NOTES-DJGPP.md
NOTES-NONSTOP.md
NOTES-PERL.md
NOTES-UNIX.md
NOTES-VALGRIND.md
NOTES-VMS.md
NOTES-WINDOWS.md
README-ENGINES.md
README-FIPS.md
README.md
README-PROVIDERS.md
SUPPORT.md
```

**Underscores**
```
ACKNOWLEDGEMENTS.md
AUTHORS.md
CHANGES.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
FAQ.md
HACKING.md
INSTALL.md
NEWS.md
NOTES_ANDROID.md
NOTES_DJGPP.md
NOTES_NONSTOP.md
NOTES_PERL.md
NOTES_UNIX.md
NOTES_VALGRIND.md
NOTES_VMS.md
NOTES_WINDOWS.md
README_ENGINES.md
README_FIPS.md
README.md
README_PROVIDERS.md
SUPPORT.md
```

**Comment by t8m** [comment]
> what is your opinion about replacing dashes with underscores (in a followup pull request)

I would see that as unnecessary churn. So no, please.

**Review by paulidale — APPROVED** [review]

**Comment by mspncp** [comment]
Merged to master in 63df86b041, thanks!

**Pull request closed without merging**